src/components/toolbar/import-buttons/hooks/use-schedule-converter.ts

```diff
@@ -26,27 +26,40 @@ export function useScheduleConverter(): UseScheduleConverterOutput {
     const workbook = new Excel.Workbook();
     workbook.xlsx.load(fileContent).then(() => {
       const sheet = workbook.getWorksheet(1);
-      const parsedFileContent = Array<Array<string>>();
-      sheet.eachRow((row, _) => {
-        parsedFileContent.push(row.values as Array<string>);
-      });
-      parsedFileContent.forEach((a) => a.shift());
-
-      while (parsedFileContent[parsedFileContent.length - 1][0] === "") {
-        parsedFileContent.pop();
-      }
 
-      setScheduleSheet(parsedFileContent);
-      if (Object.keys(parsedFileContent).length !== 0) {
-        const parser = new ScheduleParser(parsedFileContent);
-        setScheduleErrors([
-          ...parser.sections.NurseInfo.errors,
-          ...parser.sections.BabysitterInfo.errors,
-        ]);
-        setScheduleModel(parser.schedule.getDataModel());
+      try {
+        if (sheet.rowCount == 0) {
+          throw new Error("Pusty plik");
+        }
+
+        const parsedFileContent = Array<Array<string>>();
+        sheet.eachRow((row, _) => {
+          parsedFileContent.push(row.values as Array<string>);
+        });
+        parsedFileContent.forEach((a) => a.shift());
+
+        while (parsedFileContent[parsedFileContent.length - 1][0] === "") {
+          parsedFileContent.pop();
+        }
+
+        setScheduleSheet(parsedFileContent);
+        if (Object.keys(parsedFileContent).length !== 0) {
+          const parser = new ScheduleParser(parsedFileContent);
+          setScheduleErrors([
+            ...parser.sections.NurseInfo.errors,
+            ...parser.sections.BabysitterInfo.errors,
+          ]);
+          setScheduleModel(parser.schedule.getDataModel());
+        }
+      } catch (e) {
+        console.log(e.message);
+        scheduleErrors.push({
+          code: ParseErrorCode.INPUT_FILE_ERROR,
+          actual: e.message,
```

Comment:
Tak naprawdę, to mogą pojawić się nie tylko te wyjątki które wyrzucamy my. Może też być tak, że będzie `IndexOutOfRange`, `Property of undefined`, etc. 
I w obecnej implementacji my pokazalibyśmy te wyjątki użytkownikowi. Więc moim zdaniem, lepiej by było:

1. Zdefiniować wiecej pół w enumie `SectionParserError`: LOST_CHILDREN_SECTION, LOST_NURSE_SECTION, etc. 
2. Po takim zhandlowaniu jak tutaj, ustawiać `{ code : e.message }` który ty dodajesz. 
3. Zmodyfikować `ErrorMessageHelper` tak, żeby on obsługiwał te wyjątki, co będą w enumie i zwracał do każdego z nich odpowiednią wiadomość. I to miało by taki sens, że w unikamy tego problemu z wyjątkami które nie są czytelne dla nas. 

Odnośnie `Actual` - to ja źle zrobiłem. To customowe pole dla handlowania niedozwolonych symboli w grafiku Potrzeba by było zaimplementować to inaczej w taki sposób, żeby było mapowanie 1:1 -- Enum z kodami wyjątków, struktura wyjątku. ( nie koniecznie w tym  PR ). 
```typescript
export type ParserError = UnknownValueError | InputFileError;
interface UnknownValueError {
  code: ParseErrorCode.UNKNOWN_VALUE
  actual: string;
}
interface InputFileError {
  code: ParseErrorCode.INPUT_FILE_ERROR // ew. ParseErrorCode.LOST_NURSE_SECTION | ParseErrorCode.LOST_CHILDREN_SECTION ...
  missingSection: string;
}
```

I wtedy to by było mega, ponieważ dla każdego kodu wyjątku moglibyśmy mieć specyficzne pola:
![image](https://user-images.githubusercontent.com/21079319/98965503-6d626e00-250a-11eb-986c-bfaed84aebdd.png)
Zauważ, że mimo tego, że podałem typ error jako `ParserError` jako `UnknownValueError | InputFileError`, typescript wywnioskował na podstawie pola `code` o którym dokładnie interfejsie chodzi i podpowiada mi brakujące pole z  interfejsu `UnknownValueError`.

@Sig00rd , jak byś chciał, to mógłbyś też przerobić wyjątki od algorytmu na taką postać :D 
Bo kiedy mnie pytałeś o tym jak to najlepiej zrobić, to nie wiedziałem, że coś takiego istnieje.